apple/Sources/FerrostarSwiftUI/Views/InstructionsView.swift

```diff
@@ -24,72 +28,106 @@ public struct InstructionsView: View {
     ///   - visualInstruction: The visual instruction to display.
     ///   - distanceFormatter: The formatter which controls distance localization.
     ///   - distanceToNextManeuver: The distance remaining for the step.
+    ///   - remainingSteps: All steps remaining in the route, including the current step
     ///   - primaryRowTheme: The theme for the primary instruction.
     ///   - secondaryRowTheme: The theme for the secondary instruction.
-    ///   - showPillControl: If true, shows a pill control (to indicate an action/expansion).
+    ///   - sizeWhenNotExpanded: Size of the InstructionsView when minimized - useful for allocating space for this view
+    /// in your layout.
     public init(
         visualInstruction: VisualInstruction,
         distanceFormatter: Formatter = DefaultFormatters.distanceFormatter,
         distanceToNextManeuver: CLLocationDistance? = nil,
+        remainingSteps: [RouteStep]? = nil,
         primaryRowTheme: InstructionRowTheme = DefaultInstructionRowTheme(),
         secondaryRowTheme: InstructionRowTheme = DefaultSecondaryInstructionRowTheme(),
-        showPillControl: Bool = false
```

Comment:
Rather than a boolean, the pill is now shown whenever there are "next steps".

Reply:
I think it still needs to be configurable but `showPillControl` was the wrong thing to toggle ;) I'll play with a few ideas and commit something.

Reply:
Ok... I feel moderately strongly that passing parameters like this which are then promptly assigned to state is an anti-pattern in SwiftUI, no matter how tempting. (Side note: it's clear that you understand what's happening since your comment explains that this is only the initial value, but I think that puts you in the top like 2% of devs by SwiftUI knowledge :D)

I think that we can sidestep the problem by making this a binding instead of a boolean value. This way developers can still do what you intended, but with less potential confusion and _more_ flexibility:

* You can set your own state to have the default they want
* You can directly control whether it's expanded, or observe whether it is and change UI accordingly
* You can even use `.constant(true)` to force it to stay open if you really want

Ignore my comment on making the "show pill control" toggleable. You elegantly address this in the `TopDrawerView`, and you can always pass an empty set of remaining steps to this view.